src/wf.rs

```diff
@@ -134,6 +134,12 @@ impl<'a, 'lr, 'tcx> TypeChecker<'a, 'lr, 'tcx> {
         match (op.kind, &ty.kind) {
             (UnOpKind::Deref, TyKind::Ref(_, ty, _)) => ty,
             (UnOpKind::Not, TyKind::Bool) => ty,
+            // We don't need to do any unification here because the negation
+            // of an integer type is that same integer type.
+            (UnOpKind::Neg, TyKind::Int(_)) => ty,
+            (UnOpKind::Neg, TyKind::Float(_)) => ty,
+            (UnOpKind::Neg, ty::Infer(ty::IntVar(_))) => ty,
+            (UnOpKind::Neg, ty::Infer(ty::FloatVar(_))) => ty,
```

Comment:
These two cases are ok I think because in rust I'm getting the following behavior.
```rust
let x: u32 = -1; //ill-typed
```